### ISSUE TITLE
Doc Issue:Rhomobile | Runtime Application Configuration

### ISSUE THREAD

**Issue by PVT784** [rhomobile/rhomobile-docs#285]
Link: http://docs.rhomobile.com/en/5.0.25/guide/runtime_config# : 
Amy Ng, and I (Sean Kennedy) have ZenDesk ticket 3325 where the customer is requesting clarification about the documentation regarding the SSL certificates.

I find this documentation confusing as well.  

*** Begin Cut 'N Paste from 3325 ***
We have two questions regarding the documentation found at the following address:

http://docs.rhomobile.com/en/5.0.25/guide/runtime_config

It is stated that

Note: The Config.xml effects applications that are using Motorola Webkit. However the settings CAFile and CAPath in this file will be used for 4.0 applications using the stock browser.

It this meant to be interpreted such that CAFile and CAPath are used ONLY for the stock browser and NOT for Motorola Webkit? What about 5.0.25 applications? How does this apply?

In config.xml the entry example given is

<CaFile value="%INSTALLDIR%\server.pem"/>

What are the %INSTALLDIR% paths on each of the platforms (Win32, Android, iOS) and how does the pem file get put into that location?
End Cut 'N Paste ***  

If the certificate is installed along with the app in question for a RhoMobile Client install this makes sense, but the confusion occurs for users who are using 3rd party Browsers to access the Rho Content, and how they need to account for the Certificate for those too.  -- I think this is the part that causes confusion.

In addition, there is no clarity about the certificate implementation for a RhoElements Server Side controller and where the SSL certificate and configuration plays a role on the Server-Side where a customer has an In-House server providing the services.

-sean

**Comment by darryncampbell** [comment]
Hi Guys,

Gosh, that is confusing.

On Windows Mobile / CE RhoElements (Webkit) applications the CAFile parameter is used to specify the certificate to be used when making a secure connection to the server, WM / CE will NOT use the certificate store on the device, this is a limitation of our WebKit engine.

On Windowws Mobile / CE Enterprise Browser applications utilizing the Internet Explorer rendering engine, the CAFile parameter will have no effect, instead certificates loaded into the device's internal store are used.

This is all only for self signed certificates, for public sites signed by Verisign etc it is not necessary to include any certificates in CAFile.

CAPath I actually thought we had removed from the config, its implementation is confusing and should not be used

On Android, the native (stock) browser can be configured with CAFile in Config.xml or the certificates can be included in the BSP, though I'm not certain on the specifics... if this question relates to Android, please get back to me and I'll find out.

**Comment by PVT784** [comment]
Hi Darryn.
Yes, quite confusing, and for WIN32 desktops (WinXPe / WinPOS / WinXP_Pro_INDUSTRIAL) running the RhoMobile it may be closer.

The platforms in question are - Android,  iOS (which has little  no information), and WIN32. In that order for the ticket.

But it might be good as:
RhoClient  (platforms)

Native / 3rd Party web browsers.  (Local Certificates).
Little - Windows,  iOS, Android,  Other(who cares?)/Win32.


One comment for "Enterprise Android" devices -- we get many customers that wish to bypass the AOSP / Google Security constraint for "locking the terminal" when a Cert is added manually to the device. As compared to consumer devices, Zebra E/A devices can use MX to silently provision certificates, but I don't have enough detail to go into about that at this time, But I have seen Rho Tickets for that.

But if Certs are added manually to Android-- then the Lock Screen Applies -- I think it prudent to mention that for Android in particular overall.

**Comment by darryncampbell** [comment]
Hi Sean, sorry but now I'm a bit confused over what the actual question is... am I correct in saying the documentation for Android, iOS and Win32 in this area is insufficient to address the customer queries?

@rognar 's team should be able to advise on security certificates on Android, iOS and Win32, Konstantin, are those documented anywhere?

Thanks.

**Comment by PVT784** [comment]
My Apologies Darryn,  My first sentence was confusing. 

What needs to be clarified is:
RhoMobile client certificate use on 4 platforms: 1) LittleWindows, 2) iOS, 3) Android, 4) Win32/Win64.

For Native Browsers / Enterprise Browser / 3rd party browser -- it would be good to mention how those platforms do Cert Management for those clients connecting to a Rho Web Back-end.
Mention -- not provide support documentation.

My Comment about Android still stands; customers need to understand that Google drives the security settings. We (Zebra) are beholden to that.

**Comment by darryncampbell** [comment]
Hi Sean, firstly I should mention I've only just realized this github issue is under the docs folder! @michaelToews , are you aware that there was an HTTPS user guide started by @geoffday67 to start to document this sort of stuff? (https://docs.google.com/document/d/19WsJq7x2-6Rp9k73JythwW6iFAtxYuedNFKJd-sH50s/edit) [probably needs converting before it can be considered user facing].  iOS / Big Windows aren't mentioned in the doc but it's a starting point.

**Comment by PVT784** [comment]
Thanks Darryn, That's an interesting article -- it does contain some components that I'm asking for.
And correct -- iOS and BigWindows -- is missing. (Pernsonally, I will keep MacOS/BSD/Linux stuff on the shelf for now...)

I only thought of this now -- but is intrinsic to the questions in the Ticket -- how to deploy / install the Cert used with RhoMobile client.  (I assume at this point -- it's part of the RhoMobile Client installer package.) So the question relating to the %INSTALLDIR% / $INSTALLDIR variable may need a look at too.

**Comment by michaelToews** [comment]
@darryncampbell no, I had not seen that doc but it seems like something that I should have seen before...  Do you recommend a seperate doc for this info or a supplement to an existing doc?

**Comment by darryncampbell** [comment]
It would probably best be a separate doc I would say. Thanks.

**Comment by michaelToews** [comment]
Are we basically talking about this?

https://developer.motorolasolutions.com/docs/DOC-2848

**Comment by darryncampbell** [comment]
Looks good, couple of comments:
<CaFile Value=â&#128;&#157;\\mycert.pemâ&#128;&#157;>   has messed up formatting, it should be <CaFile Value="\\mycert.pem"/>
Also, the previous line mentions  ‘mycert.crt’ but should be consistent with the example which states mycert.pem.  Either is fine but would be clearer to use .pem.

Thanks.

**Comment by darryncampbell** [comment]
Also:
  ::term          openssl x509 -in cacert.pem -text -noout
Should be > rather than ::term.

@geoffday67 had a couple of other concerns.

Thanks.

**Comment by michaelToews** [comment]
Darryn, thanks for pointing out the formatting issues. That specific MD worked fine in the offline docs but I guess it didn't translate well to LP.

**Comment by michaelToews** [comment]
Darryn, this doc mentions webkit a lot because of it's affiliation with EB. Do the webkit mentions still apply?

**Comment by michaelToews** [comment]
I have added the aforementioned information to the docs but I have not yet added it into the TOC so it should not be easily accessible yet. Darryn / Sean, could you please review the new doc and give feedback on it? I also need to know if this applies to iOS as well and, if so, how as the google doc linked above has no mention of iOS.

GH Commit: https://github.com/rhomobile/rhomobile-docs/commit/dd13808f47861879a5bf6e5ee48cc73ca0bba376

**Comment by PVT784** [comment]
What you have Mike is a good start. When WebKit is referred to -- that means the Rho runtime client, or it refers to the EB, We should put a work in that the two are synomynous, and the config.xml rules.
Otherwise, the System Browser and Certificate Management of the respective platform needs the Cert wherever it needs to go for other platforms. (Safari on iOS for example)

WIN32 (BigWindows) may need special mention. But where to start / how to do it I can't say (it should be similar to LittleWindows, but...) But again, WebKit/EB/RhoClient is dictated by the config.xml and the CAFile entry so this could be a Generic mention.

**Comment by michaelToews** [comment]
Well, I don't want to refer to the webkit at all then because there is no longer any supported RhoMobile shared runtime. The alternative is Enterprise Browser and they are not to be related to one another in any way (paraphrasing Sunil and Paul Henderson). However, the config.xml is not solely used by EB / shared runtime. Native apps can use the config.xml for runtime configuration as well.

As far as Win32, I will indeed need some direction as to what to include. @darryncampbell , @rognar @dmitrys, anything you can give me as far as pointers on these items?

**Comment by darryncampbell** [comment]
Missed this: 
>>  Darryn, this doc mentions webkit a lot because of it's affiliation with EB. Do the webkit mentions still apply?
Webkit on Windows Mobile / CE is equally applicable to both Rho and EB as it's the same logic and both use Config.xml to configure the webview component.

The question on Win32 above is best answered by @rognar 's team.

Thanks

**Comment by michaelToews** [comment]
@darryncampbell ok, so the mentions of webkit are fine then. OK, I'll add those back in and wait on @rognar's team to get an answer on the Win32 side.
